src/socket.c

```diff
@@ -244,7 +255,8 @@ mrb_basicsocket_getsockname(mrb_state *mrb, mrb_value self)
 static mrb_value
 mrb_basicsocket_getsockopt(mrb_state *mrb, mrb_value self)
 { 
-  int opt, s;
+  char opt[4];
```

Comment:
These functions have different signatures on Unix and Windows:
Unix:
int getsockopt(int sockfd, int level, int optname, void *optval, socklen_t *optlen);

Windows:
int getsockopt(int s, int level, int optname, char *optval, int *optlen);

Compiling had caused me a type error because of the stricter requirement that optval be of type char* instead of any old pointer. And moreover, given that opt's only use after this was to be used as an argument for mrb_str_new, having it represented as an int makes little sense.

That said, I do notice that I am still using sizeof(int) in the mrb_str_new call, and will change this. Would you prefer that opt store at least 8 bytes instead of 4?